Sources/StreamChat/StateLayer/ChatList.swift

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An object which represents a list of `ChatChannel`.
+@available(iOS 13.0, *)
+public struct ChatList {
+    /// The query specifying and filtering the list of channels.
+    public let query: ChannelListQuery
```

Comment:
Like I said above, it feels were we have **Chat**List, but then we provic a **Channel**ListQuery
```suggestion
public struct ChannelList {
    /// The query specifying and filtering the list of channels.
    public let query: ChannelListQuery
```